src/test_c2s_srv.c

```diff
@@ -208,29 +211,39 @@ int test_c2s(int ctlsockfd, tcp_stat_agent* agent, TestOptions* testOptions,
 
       // If a valid connection request is received, client has connected.
       // Proceed.  Note the new socket fd - recvsfd- used in the throughput test
-      recvsfd = accept(testOptions->c2ssockfd,
-                       (struct sockaddr *) &cli_addr, &clilen);
-      if (recvsfd > 0) {
-        log_println(6, "accept() for %d completed",
-                    testOptions->child0);
+      c2s_conn.socket = accept(testOptions->c2ssockfd,
+                               (struct sockaddr *)&cli_addr, &clilen);
+      if (c2s_conn.socket > 0) {
+        log_println(6, "accept() for %d completed", testOptions->child0);
 
         // log protocol validation indicating client accept
         procstatusenum = PROCESS_STARTED;
         proctypeenum = CONNECT_TYPE;
         protolog_procstatus(testOptions->child0, testids, proctypeenum,
-                            procstatusenum, recvsfd);
-	// To preserve user privacy, make sure that the HTTP header
-	// processing is done prior to the start of packet capture, as many
-	// browsers have headers that uniquely identitfy a single user.
-        if (testOptions->connection_flags & WEBSOCKET_SUPPORT) {
-          if (initialize_websocket_connection(recvsfd, 0, "c2s") != 0) {
-            recvsfd = 0;
-          } 
+                            procstatusenum, c2s_conn.socket);
+        if (testOptions->connection_flags & TLS_SUPPORT) {
+          c2s_conn.ssl = SSL_new(ctx);
+          if (c2s_conn.ssl == NULL) return ENOMEM;
+          if (SSL_set_fd(c2s_conn.ssl, c2s_conn.socket) == 0) return EIO;
+          if (SSL_accept(c2s_conn.ssl) != 1) return EIO;
         }
+        // To preserve user privacy, make sure that the HTTP header
+        // processing is done prior to the start of packet capture, as many
+        // browsers have headers that uniquely identitfy a single user.
```

Comment:
That's using JavaScript, not just HTTP headers (see table on p.5 https://panopticlick.eff.org/browser-uniqueness.pdf). It would be really hard to uniquely identify a user just based on the HTTP information unless they had a custom user-agent string or a really unusual configuration.